benchmarks/ModelingToolkit/RCCircuit.jmd

```diff
@@ -0,0 +1,169 @@
+  tspan = (0.0, 10.0)
+  t0 = time()
+  prob = if target === JuliaSimCompiler.JuliaTarget()
+    ODEProblem(sys, u0, tspan, ps; sparse = true)
+  else
+    ODEProblem(sys, target, u0, tspan, ps; sparse = true)
+  end
+  (; f, u0, p) = prob
+  ff = f.f
+  du = similar(u0)
+  ff(du, u0, p, 0.0)
+  t_fode = time() - t0
+  duref === nothing || @assert duref ≈ du
+  t_run = @belapsed $ff($du, $u0, $p, 0.0)
+  t_solve = @elapsed solve(prob, Rodas5(autodiff = false))
+  (t_fode, t_run, t_solve), du
+end
+
+const C = JuliaSimCompiler.CTarget();
+const LLVM = JuliaSimCompiler.llvm.LLVMTarget();
+
+function run_and_time!(ss_times, times, max_sizes, i, n)
+  sys, u0, ps = build_system(n);
+  if n <= max_sizes[1]
+    ss_times[i, 1] = @elapsed sys_mtk = structural_simplify(sys)
+    times[i, 1], _ = compile_run_problem(sys_mtk, u0, ps)
+  end
+  ss_times[i, 2] = @elapsed sys_jsir_scalar = structural_simplify(IRSystem(sys), loop=false)
+  ss_times[i, 3] = @elapsed sys_jsir_loop = structural_simplify(JuliaSimCompiler.compressed_connection_expansion(sys))
+  oderef = daeref = nothing
+  n <= max_sizes[2] && ((times[i, 2], oderef) = compile_run_problem(sys_jsir_scalar, u0, ps; duref = oderef))
+  n <= max_sizes[3] && ((times[i, 3], oderef) = compile_run_problem(sys_jsir_scalar, u0, ps; target=C, duref = oderef))
+  n <= max_sizes[4] && ((times[i, 4], oderef) = compile_run_problem(sys_jsir_scalar, u0, ps; target=LLVM, duref = oderef))
+  n <= max_sizes[5] && ((times[i, 5], deeref) = compile_run_problem(sys_jsir_loop, u0, ps; duref = daeref))
+  n <= max_sizes[6] && ((times[i, 6], daeref) = compile_run_problem(sys_jsir_loop, u0, ps; target=C, duref = daeref))
+  n <= max_sizes[7] && ((times[i, 7], daeref) = compile_run_problem(sys_jsir_loop, u0, ps; target=LLVM, duref = daeref))
+  @show n, ss_times[i, :], times[i, :]
+end
+
+N = vcat([5, 10, 20, 40, 60, 80, 160, 320, 480, 640, 800], 1_000:1_000:10_000);
+
+# max size we test per method
+max_sizes = [1000, 4_000, 10_000, 10_000, 10_000, 10_000, 10_000];
+
+# NaN-initialize so Makie will ignore incomplete
+ss_times = fill(NaN, length(N), 3);
+times = fill((NaN,NaN,NaN), length(N), length(max_sizes));
+
+@time run_and_time!(ss_times, times, max_sizes, 1, 4); # precompile
+
+for (i, n) in enumerate(N)
+  run_and_time!(ss_times, times, max_sizes, i, n)
+end
+
+f = Figure();
+let ax = Axis(f[1, 1]; title="Structural Simplify Time")
+  for (j, label) in enumerate(("MTK", "JSIR-Scalar", "JSIR-Loop"))
+    ts = @view(ss_times[:, j])
+    lines!(N, ts; label)
+  end
+  axislegend(ax, position = :lt)
+end
+for (i, timecat) in enumerate(("ODEProblem + f!", "Run", "Solve"))
+  title = timecat * " Time"
+  ax = Axis(f[i+1, 1]; title)
+  for (j, label) in enumerate(("MTK", "JSIR - Scalar - Julia", "JSIR - Scalar - C", "JSIR - Scalar - LLVM", "JSIR - Loop - Julia", "JSIR - Loop - C", "JSIR - Loop - LLVM"))
+    ts = getindex.(@view(times[:, j]), i)
+    lines!(N, ts; label)
+  end
+  axislegend(ax, position = :lt)
```

Comment:
Copy my plot changes for moving the plot labels outside, since they cover half of the graph. https://github.com/SciML/SciMLBenchmarks.jl/pull/949/files#diff-c0aa5047a0078d961dd9b04e0cb2fac675f99d8c0b809efeb64b1ffc4bd2b8b3R398-R417

Reply:
Done.